### PR TITLE
I've largely updated the dependencies - tests seem to run, though more testing may be needed

### PR DESCRIPTION
**Pull request by bbarker**
Also, next step is to go to sbt 0.13.15, but this seems to bring up some issues with the git-based `mecha` dependency, which I'm not too sure about how to resolve immediately.

### CODE REVIEW

File: project/Build.scala [hunk 1]
```diff
@@ -23,9 +23,9 @@ object MacroGLBuild extends MechaRepoBuild {
       List("macrogl_major", "macrogl_minor"))
   }
 
-  val macroglScalaVersion = "2.11.8"
+  val macroglScalaVersion = "2.12.2"
```
Comment (axel22) [inline project/Build.scala]: We would have to cross-compile this.

**Comment by bbarker** [comment]
Tentatively added crossbuild support, but seem to caused a problem in `sbt test` to emerge that I also saw (I think) when trying about sbt 0.13.15 (though now I'm thinking the origin of the problem may not be related to sbt versions):

```
[info] Compiling 3 Scala sources to /home/brandon/.sbt/0.13/staging/44ec20cefe008edf2897/mecha/target/scala-2.10/sbt-0.13/classes...
[error] /home/brandon/.sbt/0.13/staging/44ec20cefe008edf2897/mecha/src/main/scala/org/stormenroute/mecha/package.scala:75: type mismatch;
[error]  found   : org.stormenroute.mecha.org.stormenroute.mecha.BufferedLogger
[error]  required: org.stormenroute.mecha.(some other)org.stormenroute.mecha.BufferedLogger
[error]         (name, logger)
[error]                ^
[error] /home/brandon/.sbt/0.13/staging/44ec20cefe008edf2897/mecha/src/main/scala/org/stormenroute/mecha/package.scala:87: type mismatch;
[error]  found   : org.stormenroute.mecha.org.stormenroute.mecha.BufferedLogger
[error]  required: org.stormenroute.mecha.(some other)org.stormenroute.mecha.BufferedLogger
[error]         (name, logger)
[error]                ^
[error] two errors found
[error] ({git://github.com/storm-enroute/mecha.git}mecha/compile:compileIncremental) Compilation failed```

**Comment by bbarker** [comment]
I forgot to ask in the post above - how does the project find to use mecha version 0.4-SNAPSHOT?

File: project/Build.scala [hunk 2]
```diff
@@ -173,6 +173,10 @@ object MacroGLBuild extends MechaRepoBuild {
       "-optimise",
       "-feature"
     ),
+    libraryDependencies ++= Seq(
```
Comment (axel22) [inline project/Build.scala]: This should not be needed, normally the LWJGL plugin resolves those dependencies.

File: project/Build.scala [hunk 3]
```diff
@@ -25,14 +26,15 @@ object MacroGLBuild extends MechaRepoBuild {
 
   val macroglScalaVersion = "2.11.8"
 
-  val lwjglVersion = "2.9.0"
+  val lwjglVersion = "2.9.3"
```
Comment (axel22) [inline project/Build.scala]: I would change the LWJGL version in a separate PR, since its orthogonal to adding cross-compilation.

File: project/plugins.sbt [hunk 4]
```diff
@@ -1,10 +1,13 @@
 
 resolvers += Resolver.sonatypeRepo("public")
 
-addSbtPlugin("com.typesafe.sbt" % "sbt-pgp" % "0.8.1")
+addSbtPlugin("io.get-coursier" % "sbt-coursier" % "1.0.0-M15")
+
+addSbtPlugin("com.jsuereth" % "sbt-pgp" % "1.0.0")
 
 addSbtPlugin("com.storm-enroute" % "sbt-lwjgl-plugin" % "3.1.6")
+//addSbtPlugin("com.github.philcali" % "sbt-lwjgl-plugin" % "3.1.6")
```
Comment (axel22) [inline project/plugins.sbt]: This was not published on maven under `com.github.philcali`, as far as I am aware of, so we can remove this comment, and just use the `com.storm-enroute` dependency that I published some time ago.

File: project/Build.scala [hunk 5]
```diff
@@ -94,11 +96,9 @@ object MacroGLBuild extends MechaRepoBuild {
         mechaNightlyKey.dependsOn(compile in Compile)
     )
   
-  lazy val macrogl: Project = Project(
-    "macrogl",
-    file("."),
-    settings = macroglSettings
-  )
+  lazy val macrogl: CrossProject = CrossProject("macrogl", file("."), CrossType.Full)
+    .settings(macroglSettings)
```
Comment (axel22) [inline project/Build.scala]: This seems to break the build. It is also orthogonal to 2.12 cross compilation, so I would move it to another PR.

File: project/plugins.sbt [hunk 6]
```diff
@@ -1,10 +1,13 @@
 
 resolvers += Resolver.sonatypeRepo("public")
 
-addSbtPlugin("com.typesafe.sbt" % "sbt-pgp" % "0.8.1")
+addSbtPlugin("io.get-coursier" % "sbt-coursier" % "1.0.0-M15")
```
Comment (axel22) [inline project/plugins.sbt]: Adding this to Reactors previously broke a multi-project mecha-based build (possibly because all the projects required the same sbt plugin).

File: project/plugins.sbt [hunk 7]
```diff
@@ -1,10 +1,13 @@
...
+
+addSbtPlugin("com.jsuereth" % "sbt-pgp" % "1.0.0")
```
Comment (axel22) [inline project/plugins.sbt]: This change seem orthogonal, I would suggest putting it into another PR (if it's really necessary - I generally tend to avoid changing dependencies unless it's absolutely needed).

**Comment by axel22** [comment]
I believe that the error with `BufferedLogger` happens because you are using two different versions of SBT in two different projects (possibly, these two projects are reactors and macrogl). The way that both those projects depend on the `mecha` plugin is through a Git dependency (not a Maven dependency). The reason for using this SBT's special Git dependency is that I need to develop the source of multiple projects simultaneously (including mecha), and then simultaneously push changes to multiple repositories - when I do changes in mecha, I want to immediately see them in all the projects that depend on it, and not have to go through a snapshot release (this takes time, and slows down development). Note that other people using mecha can still use the snapshot release, but since I also develop mecha, I prefer to see my changes instantly.

Now, Git dependencies in SBT have a strange behavior sometimes - namely, they are checked out from Git once for different SBT projects, and built once by one project, but then built another time by the different project, and class-loaded twice. I *think* this is the problem that you're seeing, although I never digged into it. This only seems to happen with Git deps, and since few people use them, nobody bothered to fix that in SBT.

Now, the good thing is that you should probably be able to fix this easily by cleaning this directory: `/home/brandon/.sbt/0.13/staging/44ec20cefe008edf2897/mecha/`

File: project/plugins.sbt (hunk 7 above)
Comment (bbarker) [inline project/plugins.sbt]: I was updating based on the version suggestion here: http://www.scala-sbt.org/sbt-pgp/

File: project/plugins.sbt (hunk 6 above)
Comment (bbarker) [inline project/plugins.sbt]: You mean if you use mecha, each project have to have the same exact contents in their plugins.sbt file?

File: project/Build.scala (hunk 5 above)
Comment (bbarker) [inline project/Build.scala]: Sorry, I misunderstood `CrossProject` - I thought this was necessary to support cross compilation, but I see this is just for ScalaJS.

File: project/plugins.sbt (hunk 6 above)
Comment (bbarker) [inline project/plugins.sbt]: I'll remove it for now from the PR, but I'm curious to understand the comment - I should try out mecha soon.

File: project/Build.scala (hunk 2 above)
Comment (bbarker) [inline project/Build.scala]: Removing for now - could've been a problem with another sbt lwjgl plugin version I was using.

**Comment by bbarker** [comment]
Revised PR; only thing I can recall leaving was the version bump on sbt-pgp; see http://www.scala-sbt.org/sbt-pgp/ for suggestion versions, but I'm happy with whatever you think.

**Comment by axel22** [comment]
Since I use `sbt-pgp` for a multitude of projects that need to work together, I would prefer to change that simultaneously in all these projects. Reason is, in the past, I have witnessed subtle and hard-to-notice incompatibilities between sbt plugin versions.

**Comment by bbarker** [comment]
Sounds good, reverted sbt-pgp change

**Comment by axel22** [comment]
LGTM! Thanks!
